test/classes/bradc/compilerErrorInMethod/testClear-snap1.chpl

```diff
@@ -57,5 +57,11 @@ var a = new Wrap(_value = new Assoc());
 if doDense {
   d = 1;
 }
+
 s = 1;
 a = 1;
+
+delete a._value;
+delete s._value;
+delete d._value;
```

Comment:
I had the same thought for a few tests.  In some cases the test had some code that made
it seem as if the class/record “owned” the class that had been leaked, e.g. a record that
allocated the errant class in its constructor.  However most tests, like this one, are delivering
an externally generated class as a formal to the constructor (all implied in this case) and so
I chose to make the caller be responsible for that memory.

As you say, these tests are designed to be “thoughtful” about memory management and my
goal is to remove “leak noise” from the test suite.

Mike



On Apr 20, 2016, at 9:17 AM, Ben Harshbarger <notifications@github.com<mailto:notifications@github.com>> wrote:


In test/classes/bradc/compilerErrorInMethod/testClear-snap1.chpl<https://github.com/chapel-lang/chapel/pull/3743#discussion_r60440275>:

>  s = 1;
>  a = 1;
> +
> +delete a._value;
> +delete s._value;
> +delete d._value;


While this will resolve the leaks, should they go in some sort of destructor method? It's probably OK to go in as is, since the test doesn't really intend to use memory management.

—
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub<https://github.com/chapel-lang/chapel/pull/3743/files/473eb49412a80249121b88f0ae422eb9ac39c88f#r60440275>